### PR TITLE
[Hugging Face Hub] All models to be loaded from Hugging Face Hub

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/pytorch/fairseq/blob/main/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

As discussed offline with @kahne this is a proposal of how an integration with the :hugs: Hub could look like.
The main purpose of this integration is to be able to better demo fairseq models on the :hugs: Hub for inference.
I.e. so that [`fairseqs` speech synthesis models](https://github.com/pytorch/fairseq/tree/main/fairseq/models/text_to_speech) could be easily loaded from the :hugs: hub.

This is just a rough first draft and I would be very happy any suggestions how this PR could be improved :-)

## PR review
@kahne

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
As an example, I've uploaded the [`fastspeech2` model weights](https://github.com/pytorch/fairseq/blob/main/examples/speech_synthesis/docs/ljspeech_example.md#results) to the hub [here](https://huggingface.co/patrickvonplaten/fairseq-fastspeech2).

Those weights can now be easily downloaded and cached via:

```python
from fairseq.checkpoint_utils import load_model_ensemble_and_task_from_hf

model = load_model_ensemble_and_task_from_hf("patrickvonplaten/fairseq-fastspeech2")
```

File: fairseq/checkpoint_utils.py [hunk 1]
```diff
@@ -384,6 +387,39 @@ def get_maybe_sharded_checkpoint_filename(
...
+    cache_dir: Optional[str] = None,
+    **kwargs: Any,
+):
+    LIBRARY_NAME = "fairseq"
+    CACHE_DIRECTORY = os.path.join(Path.home(), ".cache", LIBRARY_NAME)
+
+    cache_dir = cache_dir or CACHE_DIRECTORY
+
+    try:
+        from huggingface_hub import snapshot_download  # type: ignore
```
Comment (patrickvonplaten) [inline fairseq/checkpoint_utils.py]:
```suggestion
        from huggingface_hub import snapshot_download
```

**Comment by patrickvonplaten** [comment]
@kahne - this can be closed as it's already merged no? :-)

**Comment by kahne** [comment]
> @kahne - this can be closed as it's already merged no? :-)

Yes, sounds good :)

**Pull request closed without merging**